spring-xd-dirt/src/main/java/org/springframework/xd/dirt/server/admin/deployment/zk/DeploymentSupervisor.java

```diff
@@ -206,20 +207,23 @@ public DeploymentSupervisor(AdminAttributes adminAttributes) {
 	@Override
 	public void onApplicationEvent(ApplicationEvent event) {
 		if (event instanceof ContextRefreshedEvent) {
-			this.applicationContext = ((ContextRefreshedEvent) event).getApplicationContext();
-			String delay = this.applicationContext.getEnvironment().getProperty(QUIET_PERIOD_PROPERTY);
-			if (StringUtils.hasText(delay)) {
-				quietPeriod.set(Long.parseLong(delay));
-				logger.info("Set container quiet period to {} ms", delay);
-			}
-			if (this.zkConnection.isConnected()) {
-				if (this.applicationContext.equals(((ContextRefreshedEvent) event).getApplicationContext())) {
-					// initial registration, we don't yet have a port info
-					registerWithZooKeeper(zkConnection.getClient());
+			String namespace = ((EmbeddedWebApplicationContext) event.getSource()).getNamespace();
+			if (!MGMT_CONTEXT_NAMESPACE.equals(namespace)) {
+				this.applicationContext = ((ContextRefreshedEvent) event).getApplicationContext();
+				String delay = this.applicationContext.getEnvironment().getProperty(QUIET_PERIOD_PROPERTY);
+				if (StringUtils.hasText(delay)) {
+					quietPeriod.set(Long.parseLong(delay));
+					logger.info("Set container quiet period to {} ms", delay);
+				}
+				if (this.zkConnection.isConnected()) {
+					if (this.applicationContext.equals(((ContextRefreshedEvent) event).getApplicationContext())) {
```

Comment:
Looks like this is redundant. `this.applicationContext` is exactly populated with the one of the previous lines:
```
this.applicationContext = ((ContextRefreshedEvent) event).getApplicationContext();
```